lib/voight_kampff/test.rb

```diff
@@ -42,7 +45,7 @@ def preferred_path
 
     def matching_crawler
       if match = crawler_regexp.match(@user_agent_string)
-        index = match.names.first.sub(/match/, '').to_i
+        index = match.named_captures.compact.first[0].sub(/match/, '').to_i
```

Comment:
This is the actual bug. It was returning the index of the first match group 0 even when it wasn't the one that was matching. This worked fine for bot detection since any match would be true but when we started looking at properties for the crawler it broke.

Reply:
Nice catch